extensions-core/datasketches/src/main/java/io/druid/query/aggregation/datasketches/tuple/ArrayOfDoublesSketchMergeAggregator.java

```diff
@@ -0,0 +1,90 @@
+/*
+ * Licensed to Metamarkets Group Inc. (Metamarkets) under one
+ * or more contributor license agreements. See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership. Metamarkets licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package io.druid.query.aggregation.datasketches.tuple;
+
+import com.yahoo.sketches.tuple.ArrayOfDoublesSetOperationBuilder;
+import com.yahoo.sketches.tuple.ArrayOfDoublesSketch;
+import com.yahoo.sketches.tuple.ArrayOfDoublesUnion;
+import io.druid.query.aggregation.Aggregator;
+import io.druid.segment.BaseObjectColumnValueSelector;
+
+public class ArrayOfDoublesSketchMergeAggregator implements Aggregator
+{
+
+  private final BaseObjectColumnValueSelector<ArrayOfDoublesSketch> selector;
+  private ArrayOfDoublesUnion union;
+
+  public ArrayOfDoublesSketchMergeAggregator(
+      final BaseObjectColumnValueSelector<ArrayOfDoublesSketch> selector,
+      final int nominalEntries,
+      final int numberOfValues
+  )
+  {
+    this.selector = selector;
+    union = new ArrayOfDoublesSetOperationBuilder().setNominalEntries(nominalEntries).setNumberOfValues(numberOfValues)
+        .buildUnion();
+  }
+
+  /**
+   * This method is synchronized because Druid can call aggregate() and get() concurrently
+   * https://github.com/druid-io/druid/pull/3956
+   */
+  @Override
+  public synchronized void aggregate()
```

Comment:
Are locks needed in Build and Merge Aggregators alike?

Reply:
Yes. Calling get() while aggregate() is in progress can yield incorrect results.

Reply:
I mean, could this aggregator be used during realtime indexing rollup, or only during querying? If the latter, it shouldn't be synchronized. In either case, please leave corresponding comment either in class-level javadoc or javadocs of the synchronized methods, so that a reader of this class couldn't have a question like the one that I asked.